tests/tracer/test_trace_utils.py

```diff
@@ -414,7 +418,7 @@ def test_set_http_meta(
             tag = "http.request.headers." + header
             assert span.get_tag(tag) == value
 
-    if appsec_enabled:
+    if appsec_enabled and span.span_type == SpanTypes.WEB:
```

Comment:
Should this bug fix be pulled out separately and backported?

Reply:
it did not have any impact previously as we were rewriting context at the end of the span. That's why it was not detected.

Reply:
But I tried to optimize things better so we are not rewriting data at the end of the span if it was already done by a previous call to the WAF. That's why it was detected only now.